lib/chewy/type/mapping.rb

```diff
@@ -118,6 +119,35 @@ def field *args, &block
           end
         end
 
+        # Defines an aggregation that can be bound to a query or filter
+        #
+        #   Suppose that a user has posts and each post has ratings
+        #   avg_post_rating is the mean of all ratings
+        #
+        #   class UsersIndex < Chewy::Index
+        #     define_type User do
+        #       field :posts do
+        #         field :rating
+        #       end
+        #
+        #       agg :avg_rating do
+        #         { avg: { field: 'posts.rating' } }
+        #       end
+        #     end
+        #   end
+        def agg *args, &block
+          options = args.extract_options!
+          build_root unless root_object
+
+          self._agg_defs ||= {}
+          _agg_defs[args.first] = block
```

Comment:
Well, it still modifies parent hash, you have to do `self._agg_defs = _agg_defs.merge(args.first => block)`
And also move `self._agg_defs = {}` near `class_attribute :_agg_defs`. And probably you don't need `def agg_defs` anymore